orchestrator/services/watch_rerun.py

```
@@ -0,0 +1,598 @@
+def estimate_rerun_cost_usd(
+    db: Session, workspace_id: UUID | str, execution_id: str
+) -> float:
+    """What the original execution cost -- the honest rerun estimate."""
+    try:
+        from services.report_service import compute_execution_metrics
+
+        metrics = compute_execution_metrics(
+            db, workspace_id, execution_id=str(execution_id)
+        )
+        return float(metrics.get("cost_usd") or 0.0)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)